### PR TITLE
super should not consider parents without a body

### PR DESCRIPTION
**Pull request by seanyoung**
If the parent is an interface, don't use a function from that for super, as it is uncallable and will generate bogus code.

Fixes: https://github.com/xermicus/fuzzy-sol/issues/128

Signed-off-by: Sean Young <sean@mess.org>

### CODE REVIEW

**Review by xermicus — APPROVED** [review]

File: tests/substrate_tests/inheritance.rs [hunk 1]
```diff
@@ -645,6 +645,35 @@ fn test_super() {
...
+            function foo(uint64 x) public virtual returns (uint64) {
+                return 202 + x;
+            }
+        }"##,
+    );
+
+    runtime.constructor(0, Vec::new());
+    runtime.function("bar", Vec::new());
+
+    assert_eq!(runtime.vm.output, 212u64.encode());
```
Comment (LucasSte) [inline tests/substrate_tests/inheritance.rs:676]: Why are you not using `BorshToken` here?

Comment (LucasSte) [reply tests/substrate_tests/inheritance.rs:676]: That means checking the returns of `runtime.function`;

Comment (seanyoung) [reply tests/substrate_tests/inheritance.rs:676]: These are the Substrate tests

Comment (LucasSte) [reply tests/substrate_tests/inheritance.rs:676]: I totally missed that.

**Review by LucasSte — APPROVED** [review]

**Pull request merged**